src/ast/nodes/Identifier.ts

```diff
@@ -1,36 +1,32 @@
 /// <reference path="./Identifier.d.ts" />
 
-import Node, { ForEachReturnExpressionCallback } from '../Node';
+import { Node } from './shared/Node';
 import isReference from 'is-reference';
-import { UNKNOWN_ASSIGNMENT, UnknownAssignment, PredicateFunction } from '../values';
+import { UNKNOWN_EXPRESSION } from '../values';
 import Scope from '../scopes/Scope';
-import Expression from './Expression';
 import ExecutionPathOptions from '../ExecutionPathOptions';
-import Declaration from './Declaration';
 import Variable from '../variables/Variable';
 import CallOptions from '../CallOptions';
 import FunctionScope from '../scopes/FunctionScope';
 import MagicString from 'magic-string';
 import Property from './Property';
 import { ObjectPath } from '../variables/VariableReassignmentTracker';
+import { GenericExpressionNode, Expression, ForEachReturnExpressionCallback, SomeReturnExpressionCallback } from './shared/Expression';
+import { PatternNode } from './shared/Pattern';
 
-export default class Identifier extends Node {
+export function isIdentifier (node: Node): node is Identifier {
+	return node.type === 'Identifier';
```

Comment:
I am seeing a lot of these string running around.

Wouldn't it make sense to have a `string enum` instead, if you really don't want to have numeric representation.

Example;

```ts
export enum NodeType {
    Identifier = "Identifier",
    BlockElement = "BlockElement"
    ...
}
```

And than from here for instance you'd be able to use 

```ts
return node.type === NodeType.Identifier;
```

The nice thing about this, if that refactors would be easier and even finding symbol references etc...

Reply:
That makes sense, I'll add this.

Reply:
If wanting to switch to an enum here, perhaps consider going for the numeric form? For example add to enhance:

```js
node.nodeType = NodeType[node.type];
```

Then the is check would be the same as the above.

Reply:
Actually even better than the above would be to add the type to the prototype of the Node type itself:

```js
class Identifier {
}
Identifier.prototype.nodeType = NodeType.Identifier;
```

That would avoid adding to the memory footprint of Nodes or having anything in enhance, while getting the benefit of a numeric enum.

Reply:
I am not a seasoned TypeScript expert but to me it looks like there are no advantages of using a numeric enum over a const string enum which has no overhead except that the code base might be just a few bytes smaller. Acorn is already adding the right string constants to all nodes anyway. I also could not find any information about performance benefits of one over the other. Unless there is some hard argument for using numeric enums I would go for a const enum with strings (which is basically the original code)